src/lib/MessageStream.ts

```diff
@@ -146,21 +148,24 @@ export class MessageStream implements AsyncIterable<MessageStreamEvent> {
     for (const message of params.messages) {
       runner._addMessageParam(message);
     }
-    runner._run(() =>
-      runner._createMessage(
-        messages,
-        { ...params, stream: true },
-        { ...options, headers: { ...options?.headers, 'X-Stainless-Helper-Method': 'stream' } },
-      ),
-    );
+    void runner
+      ._run(
+        async () =>
+          await runner._createMessage(
```

Comment:
async function with a single await in the body is equivalent to the previous behavior; you should be able to just return the promise directly

```suggestion
    runner._run(() =>
      runner._createMessage(
```